src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1885,6 +1926,140 @@ class module_sram : public module_impl
     return payload;
   }
 
+  struct dlib_handle
+  {
+    void* handle = nullptr;
+
+    dlib_handle(const std::string& path)
+    {
+      handle = xrt_core::dlopen(path.c_str(), RTLD_LAZY);
+    }
+
+    ~dlib_handle()
+    {
+      if (handle)
+        xrt_core::dlclose(handle);
+    }
+  };
```

Comment:
This is marginally silly, but you may want to consider hiding the unique_ptr within struct dlib_handle.   Take a look at runtime_src/core/common/runner/cpu.cpp:dllwrap.

If this is temporary code based on AIEBU comment below, then please ignore this comment.

Reply:
It is temporary code but I will refactor it as code of dllwrap makes more sense, Thanks Soren.